custom_reg_form/management/commands/email_opt_in_list_to_s3.py

```diff
@@ -0,0 +1,257 @@
+
+def construct_query(fields):
+    """Returns a string `query` to be passed to `cursor.execute`"""
+    reg_ext_table = get_registration_ext_model()._meta.db_table # pylint: disable=protected-access
+    field_names = ", ".join(
+        [".".join([infer_field_model(field)._meta.db_table, field]) for field in fields]    # pylint: disable=protected-access
+    )
+    query = """
+        SELECT 
+            {field_names}
+        FROM auth_user 
+            LEFT JOIN {reg_ext_table} ON {reg_ext_table}.user_id = auth_user.id
+            JOIN auth_userprofile ON auth_userprofile.user_id = auth_user.id;
+        """.format(
+            field_names=field_names, reg_ext_table=reg_ext_table
+        )
+
+    return query
+
+
+class Command(BaseCommand):
+    """
+    Admin command to export User/UserProfile/Registration Extension fields to csv file on s3
+    """
+
+    help = "Exports users email opt-in preferences to an s3 bucket"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--use-temp-file",
+            action="store_true",
+            help=(
+                "Use a temp file instead of keeping the whole csv in memory,"
+                "useful for when expecting a large data set"
+            ),
+        )
+        parser.add_argument(
+            "--skip-when-null",
+            dest="skip_when_null",
+            action="store_true",
+            help=(
+                "when enabled the script skips the row when the value of"
+                " one of the fields is null (default=True)"
+            )
+        )
+        parser.add_argument(
+            "--no-skip-when-null", dest="skip_when_null", action="store_false"
+        )
+        parser.set_defaults(skip_when_null=True)
+        parser.add_argument(
+            "--aws-access-key-setting",
+            type=str,
+            help="specify which setting to use as an aws access key",
+        )
+        parser.add_argument(
+            "--aws-secret-key-setting",
+            type=str,
+            help="specify which setting to use as an aws secret key",
+        )
+        parser.add_argument(
+            "--aws-profile", type=str, help="specify which aws profile to use"
+        )
+        parser.add_argument(
+            "--fields",
+            type=str,
+            nargs="+",
+            help="fields to include in the output, see docstring header for how to specify fields",
+        )
+        parser.add_argument(
+            "--usage", action="store_true", help="print the module docstring header"
+        )
+        parser.add_argument(
+            "--s3-bucket-name",
+            type=str,
+            help="aws s3 bucket name where the csv file will be saved.",
+        )
+        parser.add_argument(
+            "--s3-object-name",
+            type=str,
+            help="the name under which the output will be saved.",
+        )
+
+    def handle(self, *args, **options):
+
+        if options["usage"]:
```

Comment:
I believe that the management command can generate the usage info for itself when using the `--help` flag, so this may be not needed at all.

Reply:
It is used to prompt the user to read some information about some implementation details, it is optional, and the --help is implemented to give a listing of options and their descriptions.